packages/atoms/cal-provider/index.tsx

```diff
@@ -1,15 +1,45 @@
 import type { ReactNode } from "react";
-import { createContext, useContext } from "react";
+import { createContext, useContext, useState, useCallback, useEffect } from "react";
 
 type CalProviderProps = {
   apiKey: string;
   children: ReactNode;
 };
 
-const ApiKeyContext = createContext("");
+const ApiKeyContext = createContext({ key: "", error: "" });
```

Comment:
Created two different states, one for api key values in case of error messages based on which every atom can do proper error handling